### ISSUE TITLE
TypeError: must be list, not bytearray

### ISSUE THREAD

**Issue by kdheepak89** [ManiacalLabs/BiblioPixel#41]
I get the following error when I try to run an animation test code

```
pi@raspberrypi:~/GitRepos/BiblioPixel $ sudo python run.py
INFO - spi_driver_base - py-spidev speed @ 2.0 MHz
Traceback (most recent call last):
  File "run.py", line 19, in <module>
    led.update()
  File "/home/pi/GitRepos/BiblioPixel/bibliopixel/led.py", line 136, in update
    d._update(self.buffer[pos:d.bufByteCount + pos])
  File "/home/pi/GitRepos/BiblioPixel/bibliopixel/drivers/driver_base.py", line 51, in _update
    self.update(data)
  File "/home/pi/GitRepos/BiblioPixel/bibliopixel/drivers/spi_driver_base.py", line 80, in update
    self._sendData()
  File "/home/pi/GitRepos/BiblioPixel/bibliopixel/drivers/spi_driver_base.py", line 73, in _sendData
    self.spi.xfer2(self._buf)
TypeError: must be list, not bytearray
pi@raspberrypi:~/GitRepos/BiblioPixel $
```

It looks like SPI expects a list and not a bytearray

The results of `pip freeze` are below :

```
RPi.GPIO==0.6.0a3
RTIMULib==7.2.1
RasPiLEDStrip==0.9b
Werkzeug==0.11.3
argparse==1.2.1
chardet==2.3.0
colorama==0.3.2
gpiozero==1.0.0
html5lib==0.999
itsdangerous==0.24
lxkeymap==0.1
mcpi==0.1.1
ndg-httpsclient==0.3.2
numpy==1.8.2
picamera==1.10
pifacecommon==4.2.1
pifacedigitalio==3.1.0
pyOpenSSL==0.13.1
pyasn1==0.1.7
pygame==1.9.2a0
pygobject==3.14.0
pyserial==3.0.1
requests==2.4.3
sense-hat==2.1.0
six==1.8.0
spidev==3.2
urllib3==1.9.1
wheel==0.24.0
wsgiref==0.1.2
```

**Comment by adammhaile** [comment]
Can you attach your code (run.py)?

**Comment by adammhaile** [comment]
Ahh... spidev was recently updated which made a breaking change.
It will take me a few days to make and test a fix. In the meantime, install spidev 3.1 and you'll be fine.

**Comment by kdheepak89** [comment]
Thanks for the quick reply! I used the example in the Wiki for the animation test of LEDStrip.

I found that commenting out [this line](https://github.com/ManiacalLabs/BiblioPixel/blob/master/bibliopixel/drivers/spi_driver_base.py#L21) solved the issue. Unfortunately, I wasn't able to find a version in spidev e.g. `spidev.__version__`. If there were a way to check, then we could ensure backward compatibility. We could use a `try: ... except: ...` block to determine if we should create a bytearray or list, but I'm not sure if that's an elegant solution.

**Comment by adammhaile** [comment]
I've got a message out to the dev of spidev. Back in the spidev 2.x days (when I first wrote bibliopixel) you HAD to use a bytearray for xfer2. But it seems that in v3.0 and v3.1 it started to accept lists and was accidentally still working with bytearrays. Then, on March 7th he submitted a patch that makes it fail if it's *not* a list: https://github.com/doceme/py-spidev/commit/bbe43739bf42410739684a5c65a1d8b4723f25f6#diff-fed16b093906477445b903aa554b610d

That was released in spidev 3.2, which you have. So, yeah... I'm probably going to have to only run the line you mentioned *if* spidev version is < 3. I'm just waiting to confirm from @docme

But you are also correct that spidev doesn't have the version in the class. It's only in the setup.py file for the package... which is lame. I'd submit a bug, but it doesn't really help since old versions wouldn't have it.

Will require some testing with various versions... will do so in the next couple days.

**Comment by kdheepak89** [comment]
Thanks again for the quick follow up! I suspect it is a Python 3 compatibility issue.

**Comment by adammhaile** [comment]
Yeah... maybe so. spidev v3 was mainly to support Python 3.
The first thing I'm going to try is if a regular list works with the spidev 2.x series. If so, I'll just remove that line. Can't remember why I needed a bytearray in the first place :P

**Comment by kdheepak89** [comment]
Haha that will be interesting to see! Let me know if I can help in any way!

**Comment by adammhaile** [comment]
Fixed as of BiblioPixel v2.0.10
